src/core/lib/iomgr/wakeup_fd_eventfd.c

```diff
@@ -45,11 +45,10 @@
 #include "src/core/lib/profiling/timers.h"
 
 static grpc_error* eventfd_create(grpc_wakeup_fd* fd_info) {
-  int efd = eventfd(0, EFD_NONBLOCK | EFD_CLOEXEC);
-  if (efd < 0) {
+  fd_info->read_fd = eventfd(0, EFD_NONBLOCK | EFD_CLOEXEC);
+  if (fd_info->read_fd < 0) {
     return GRPC_OS_ERROR(errno, "eventfd");
   }
-  fd_info->read_fd = efd;
   fd_info->write_fd = -1;
```

Comment:
So I checked the codebase and It probably wouldn't impact fixing this issue, but it would be better to move it to before the if statement. Not requiring this change since this is a pre-existing problem, but you can move it if you want.